### PR TITLE
fixing syncing issue

### PR DESCRIPTION
**Pull request by OmerBenHayun**
This PR continues #4

### CODE REVIEW

File: handler/handler.go [hunk 1]
```diff
@@ -223,6 +224,20 @@ func newHandler(fn interface{}) (Func, error) {
 			}
 			return out, nil, nil
 		}
+	default:
+		// A function that returns a value,function and an error.
```
Comment (roytman) [inline handler/handler.go]: A function that returns a value, a callback function and an error.

File: handler/handler.go [hunk 2]
```diff
@@ -223,6 +224,20 @@ func newHandler(fn interface{}) (Func, error) {
...
+		decodeOut = func(vals []reflect.Value) (interface{}, func(), error) {
+			out,ofunc, oerr := vals[0].Interface(), vals[1].Interface(), vals[2].Interface()
```
Comment (roytman) [inline handler/handler.go]: out, ofunc, oerr

File: handler/handler.go [hunk 3]
```diff
@@ -223,6 +224,20 @@ func newHandler(fn interface{}) (Func, error) {
...
+			if oerr != nil {
+				return nil, nil, oerr.(error)
+			}
+			var f func()
+			f,ok := ofunc.(func())
```
Comment (roytman) [inline handler/handler.go]: f, ok

File: handler/handler.go (hunk 1 above)
Comment (OmerBenHayun) [reply handler/handler.go]: done.Thanks.

File: handler/handler.go (hunk 2 above)
Comment (OmerBenHayun) [reply handler/handler.go]: done.Thanks.

File: handler/handler.go (hunk 3 above)
Comment (OmerBenHayun) [reply handler/handler.go]: done.Thanks.

**Pull request merged**